cmd/goal/clerk.go

```diff
@@ -221,8 +221,7 @@ func waitForCommit(client libgoal.Client, txid string, transactionLastValidRound
 		}
 
 		reportInfof(infoTxPending, txid, stat.LastRound)
-		// WaitForRound waits until round "stat.LastRound+1" is committed
-		stat, err = client.WaitForRound(stat.LastRound)
+		stat, err = client.WaitForRound(stat.LastRound + 1)
```

Comment:
Not sure why this is related to heartbeats, but it seems like the handler of WaitForRound (under `/v2/status/wait-for-block-after/`) already adds +1, so this seems unnecessary.  (I don't know why that other +1 is necessary either.)

https://github.com/algorand/go-algorand/blob/master/daemon/algod/api/server/v2/handlers.go#L1065

Reply:
Oh, I see, this PR also changes that handler.  Fixing handler semantics seems reasonable, though we might end up in some busy-waiting loops when running an old goal binary against a new algod.  Nothing should go wrong other than wasting some CPU time, though.